python/cudf/cudf/tests/test_avro.py

```diff
@@ -49,16 +49,22 @@ def _make_path_or_buf(src):
 @pytest.mark.parametrize("inputfile, columns", [("example.avro", None)])
 def test_avro_reader_basic(datadir, inputfile, columns, engine):
     path = datadir / inputfile
+
+    got = cudf.read_avro(path, engine=engine, columns=columns)
+
     try:
-        reader = fa.reader(open(path, "rb"))
+        avro_file = open(path, "rb")
+        reader = fa.reader(avro_file)
+
     except Exception as excpr:
         if type(excpr).__name__ == "FileNotFoundError":
             pytest.skip(".avro file is not found")
         else:
             print(type(excpr).__name__)
 
     expect = pd.DataFrame.from_records(reader)
-    got = cudf.read_avro(path, engine=engine, columns=columns)
+    if not avro_file.closed:
+        avro_file.close()
```

Comment:
@millerhooks you should be able to remove all of these changes now since we identified the issue as being in the cudf reader as opposed to in fastavro.

Reply:
Excellent.

Reply:
Pushed. I suspect that if we have actually solved the problem then then 

`assert_eq(expect, got[expect.columns], check_categorical=False)`

No longer needs `[expected.columns]`. That was an artifact from misinterpreting this issue. I pulled from `branch-0.12` and it did not have the `xfail` that was required in #3741. I if this doesn't run, then I guess I can add the `xfail` back?